pkg/k8s/types/slim/client/clientset/versioned/clientset.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2017-2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Code generated by client-gen. DO NOT EDIT.
```

Comment:
Not related to the contents of this PR, just something that puzzled me during review:

I was wondering why GitHub didn't recognize this file as being auto-generate despite this line being present while other files with the same comment are correctly recongnized. Turns out that `github-linguist` only checks the first 10 lines of a file for the `Code generated by` line:

https://github.com/github/linguist/blob/309acde92e04c86466441afc12992274be938473/lib/linguist/generated.rb#L305

However, https://golang.org/s/generatedcode states that the comment may appear anywhere in the file. I wonder whether it'd be possible to improve `github-linguist` in that regard /cc @pchaigno

Reply:
You're right, Linguist performs a best-effort detection of generated files in that regard. We impose a limit because we want to avoid searching potentially huge files. Nevertheless, in the present case, I think the limit could be bumped to match the longest license text that could be included here (probably BSD 4-clause, so about 40 lines).

If you want to send a quick fix for that, I can make sure it gets included in the next release (usually once a month).

Reply:
Thanks for elaborating @pchaigno. I agree that it wouldn't make sense to search the whole file. Will send a quick fix bumping the limit.

Reply:
Sent PR github/linguist#4857

Reply:
We could also switch to SPDX to minimize boilerplate and not have to search so much :-)

Reply:
> We could also switch to SPDX to minimize boilerplate and not have to search so much :-)

Please do, so we have it for our code-base everywhere, not just C code. :-)